lib/update-check.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+var latestVersion = require('latest-version');
+var semver = require('semver');
+var info = require('../package.json');
+var settings = require('../settings');
+var Spinner = require('cli-spinner').Spinner;
+var chalk = require('chalk');
+Spinner.setDefaultSpinnerString(Spinner.spinners[7]);
+
+function check(next) {
+	console.error('particle-cli v' + info.version);
+	console.error();
+
+	if (settings.disableUpdateCheck) {
+		return next();
+	}
+
+	var now = Date.now();
+	var lastCheck = settings.profile_json.last_version_check || 0;
+	// check at most once an hour
+	if (now - lastCheck >= (60 * 60 * 1000)) {
+		settings.profile_json.last_version_check = now;
+		checkVersion(now, function () {
+			settings.saveProfileData();
+			start(next);
+		});
+		return;
+	}
+
+	start(next);
+}
+
+function checkVersion(now, next) {
+	var spin = new Spinner('Checking for updates...');
+	spin.start();
+	latestVersion(info.name).then(function (version) {
```

Comment:
Can you add a timeout here? You could `race` 2 promises, one that just rejects after a timeout and the one returned by `latestVersion`. I'm thinking what would happen if you're not connected to the internet and you run the CLI.